### ISSUE TITLE
Support shortcutkeys to save !

### ISSUE THREAD

**Issue by dnknn** [allen505/save-for-later-firefox#12]
<img align=right width=38% src=https://user-images.githubusercontent.com/10995302/108430045-bc434080-727b-11eb-8c13-fd59954acdc4.png>

	`chrome://extensions/shortcuts#:~:text=Save%20for%20later`

↑ Copy the above URL and open it will arrive directly. 
this hotkey is Popup, not a specific function , so must  add  ⌨hotkey for 👉

---

- Users will be able to use shortcutkey to directly trigger the function.
- Use the hotkey, it must should 🔔`Chrome.Notifications` !
https://developer.chrome.com/docs/extensions/reference/notifications/

　
　
With the 🔔, users can rest assured when saving with ⌨hotkey!

![bell](https://user-images.githubusercontent.com/10995302/108431836-66bc6300-727e-11eb-9b5f-ee035cdf3bdb.gif)![212_19064108](https://user-images.githubusercontent.com/10995302/108431147-75eee100-727d-11eb-8d17-5c4773bb26df.png)![bell](https://user-images.githubusercontent.com/10995302/108431836-66bc6300-727e-11eb-9b5f-ee035cdf3bdb.gif)

- ### 　　　　`?` tabs be saved ꘌ see at here👉 https://github.com/allen505/save-for-later/issues/13

---

⚠ Of course, this hotkey operating must also support  `Save selected tabs` !
In fact, a hotkey will be enough, because you holding the `Shift`  on the tab-bar and click 2 times from the beginning to the end, it is equivalent to selecting all the tabs under the window.

 - see here 👉 https://github.com/allen505/save-for-later/issues/13

**Comment by dnknn** [comment]
@allen505 
The pictures and decorations have been updated, and this suggestion is now perfect 🤦!

**Comment by allen505** [comment]
Hi @dnknn . This is a nice suggestion.  
However I am not prioritizing this at the moment because there are a lot of features in the pipeline for the next version already. I'm also not entirely sure if a notification is required when a save is performed. Maybe we can add that as an option "_Notify when a Save is made using shortcut_ : On/Off"
References:
[Adding a keyboard shortcut for the extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/commands)
[Adding a notification](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Notifications)

**Comment by dnknn** [comment]
> I'm also not entirely sure if a notification is required when a save is performed.

Of course, no notification is required for the actions of users clicking with the mouse on Popup. Because the operation is directly visible!

But using hotkey is different.

1. Save, obviously it is an important operation.
2. If no notification, it will be silent. e.g. Press the wrong hotkey, e.g. it was not successfully saved, etc.
3. Obviously, this kind of notification is just a perceptual reminder, so it will automatically disappear/close after a few seconds! 

Therefore, this notification behavior is very reasonable and necessary, and will not burden the user (the notification will automatically close after a few seconds)

> Maybe we can add that as an option "_Notify when a Save is made using shortcut_ : On/Off"

it's up to you!

**Comment by allen505** [comment]
My reasoning is that notification will require an extra permission and is system generated AFAIK. I was thinking something like a small popup from the extension's button or a maybe a text overlay. Need to do a bit more research on notification before finalizing

**Comment by dnknn** [comment]
@allen505 

> notification will require an extra permission

There is no problem at all, because notification itself is a very basic function. When you need it, you must have notification permissions. If you use enough extensions, you can obviously feel that many extensions have notifications, of course, the premise is they It does need to be notified.

![212_20111147](https://user-images.githubusercontent.com/10995302/108582181-731ee980-736c-11eb-917c-347fcfb221ba.png)

I know what you mean, it's okay if you don't extension much, but if you extension a lot, do you think you can see it?

Even the extension icon is hidded, so your approach is useless.

So when using hotkeys to assist with notifications is the best way!

**Comment by allen505** [comment]
I see your point. I'll update here after a better research on implementation details
